MapboxNavigationTests/NavigationViewControllerTests.swift

```diff
@@ -0,0 +1,192 @@
+            XCTAssertEqual(currentRoadName, roadName, "Expected: \(roadName); Actual: \(currentRoadName)")
+            XCTAssertFalse(wayNameView.isHidden, "WayNameView should be visible.")
+            
+            mapviewExpectation.fulfill()
+        }
+        
+        wait(for: [mapviewExpectation], timeout: delay)
+    }
+    
+    // Brief: navigationViewController(_:roadNameAt:) delegate method is implemented,
+    //        with a blank road name (empty string) provided and wayNameView label is hidden.
+    
+    func testNavigationViewControllerDelegateRoadNameAtLocationEmptyString() {
+        
+        let navigationViewController = dependencies.navigationViewController
+        let routeController = navigationViewController.routeController!
+        
+        // Identify a location to set the custom road name.
+        let turkStreetLocation = dependencies.poi[1]
+        let roadName = ""
+        customRoadName[turkStreetLocation.coordinate] = roadName
+        
+        let mapviewExpectation = expectation(description: "mapview_exp_hide_roadname")
+        let customMapView = MGLMapView(frame: (navigationViewController.mapView?.bounds)!, styleURL: URL(string: "mapbox://styles/mapbox/streets-v9"))
+        
+        let delay = 10.0
+        let fireTime = DispatchTime.now() + delay
+        
+        DispatchQueue.main.asyncAfter(deadline: fireTime) {
+            
+            navigationViewController.mapView?.delegate?.mapView?(customMapView, didFinishLoading: customMapView.style!)
+            navigationViewController.mapView?.delegate?.mapViewDidFinishLoadingMap?(customMapView)
+            
+            routeController.locationManager(routeController.locationManager, didUpdateLocations: [turkStreetLocation])
+            
+            let wayNameView = (navigationViewController.mapViewController?.navigationView.wayNameView)!
+            let currentRoadName = wayNameView.text!
+            XCTAssertEqual(currentRoadName, roadName, "Expected: \(roadName); Actual: \(currentRoadName)")
+            XCTAssertTrue(wayNameView.isHidden, "WayNameView should be hidden.")
+            
+            mapviewExpectation.fulfill()
+        }
+        
+        wait(for: [mapviewExpectation], timeout: delay)
+    }
+    
+    // TODO: Most likely a UI-Test candidate.
+    func testNavigationViewControllerDelegateRoadNameAtLocationUmimplemented() {
+        
+        let dummyNavigationViewController = dependencies.dummyNavigationViewController
+        let routeController = dummyNavigationViewController.routeController!
+        
+        // Identify a location without a custom road name.
+        let fultonStreetLocation = dependencies.poi[2]
+        
+        let mapviewExpectation = expectation(description: "mapview_exp_default_roadname")
+        let customMapView = MGLMapView(frame: (dummyNavigationViewController.mapView?.bounds)!, styleURL: URL(string: "mapbox://styles/mapbox/streets-v9"))
```

Comment:
RouteMapViewController’s road name label functionality is powered by business logic within that class, dependent on a NavigationMapView that it sets up. Here, we’re creating a new, plain-vanilla MGLMapView, which could deviate significantly from how the NavigationMapView is normally set up inside RouteMapViewController.